core/src/actors/chain_manager/mod.rs

```diff
@@ -246,6 +225,35 @@ impl ChainManager {
             .wait(ctx)
     }
 
+    /// Method to Send an Item to Inventory Manager
+    fn persist_data_request(
+        &self,
+        ctx: &mut Context<Self>,
+        data_request: &(OutputPointer, DataRequestReport),
+    ) {
+        // Get StorageManager address
+        let storage_manager_addr = System::current().registry().get::<StorageManager>();
+
+        // Persist block_chain into storage. `AsyncContext::wait` registers
+        // future within context, but context waits until this future resolves
+        // before processing any other events.
+        let msg = Put::from_value(data_request.0.to_bytes().unwrap(), &data_request.1).unwrap();
```

Comment:
AFAIK, you can deconstruct `data_request` in the function signature so that you save some nasty index accessor:
```rust
(output_pointer, data_request_report): &(OutputPointer, DataRequestReport),
```
```rust
let msg = Put::from_value(output_pointer.to_bytes().unwrap(), &data_request_report).unwrap();
```

Reply:
I love it!